src/components/Calculator.js

```diff
@@ -1,55 +1,256 @@
 import React from 'react';
-import CalButton from './NumberButton';
+import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 import './Calculator.css';
-import OperationButton from './OperationButton';
-import Output from './Output';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleButtonClick = this.handleButtonClick.bind(this);
+  }
+
+  handleButtonClick(e) {
+    const handleButton = { x: '', y: '' };
+    handleButton.x = '';
+    handleButton.y = e.target.innerText;
+    let { total, next, operation } = this.state;
+    const data = calculate({ total, next, operation }, e.target.innerText);
+    total = data.total;
+    next = data.next;
+    operation = data.operation;
+    this.setState({ total, next, operation });
+
+    const { handleButtonClick } = this.props;
+    if (total !== null && next === null && operation === null) {
+      const x = 'clear';
+      const y = total;
+      handleButtonClick({ x, y });
+    } else if (total === null && next && next.length === 1 && operation === undefined) {
+      const x = 'old';
+      const y = next;
+      handleButtonClick({ x, y });
+    } else {
+      const x = '';
+      const y = e.target.innerText;
+      handleButtonClick({ x, y });
+    }
   }
 
   render() {
+    const { output } = this.props;
     return (
       <div className="Calculator">
         <div className="Calculator-Ctn">
           <div className="Calculator-Row">
-            <Output output="0" />
+            <span className="output">{output}</span>
           </div>
-          <div className="Calculator-Row">
-            <CalButton output="AC" />
-            <CalButton output="+/-" />
-            <CalButton output="%" />
-            <OperationButton output="÷" />
+          <div className="Calculator-Row num-op">
+            <div
+              role="button"
+              tabIndex={0}
+              className="btn"
+              onClick={this.handleButtonClick}
+              onKeyDown={this.handleClick}
+            >
+              <span>AC</span>
```

Comment:
- [ ] This button `AC` has to return to the default zero it is not just letters please address that